evergreen/config.yml

```diff
@@ -1322,173 +1239,132 @@ task_groups:
+  display_name: "Ubuntu (Release build)"
+  run_on: ubuntu2204-arm64-large
   expansions:
-    clang_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/clang%2Bllvm-11.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz"
-    cmake_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/cmake-3.26.3-linux-x86_64.tar.gz"
-    bloaty_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/bloaty-v1.1-39-gefc1c61-ubuntu2004-x86_64.tar.gz"
-    cmake_bindir: "./cmake_binaries/bin"
     cmake_build_type: RelWithDebInfo
     fetch_missing_dependencies: On
-    c_compiler: "./clang_binaries/bin/clang"
-    cxx_compiler: "./clang_binaries/bin/clang++"
+    c_compiler: "/opt/clang+llvm/bin/clang"
+    cxx_compiler: "/opt/clang+llvm/bin/clang++"
     python3: /opt/mongodbtoolchain/v3/bin/python3
   tasks:
   - name: compile_test_and_package
-    distros:
-    - ubuntu2004-large
   - name: benchmarks
-    distros:
-    - ubuntu2004-large
   - name: long-running-tests
-  - name: bloaty
 
-- name: ubuntu2004-asan
-  display_name: "Ubuntu 20.04 x86_64 (Clang 11 ASAN)"
-  run_on: ubuntu2004-large
+- name: ubuntu-asan
+  display_name: "Ubuntu (ASAN ARM64)"
+  run_on: ubuntu2204-arm64-large
   expansions:
-    clang_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/clang%2Bllvm-11.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz"
-    cmake_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/cmake-3.26.3-linux-x86_64.tar.gz"
-    cmake_bindir: "./cmake_binaries/bin"
     fetch_missing_dependencies: On
     enable_asan: On
     cmake_build_type: RelWithDebInfo
-    c_compiler: "./clang_binaries/bin/clang"
-    cxx_compiler: "./clang_binaries/bin/clang++"
-    llvm_symbolizer: "./clang_binaries/bin/llvm-symbolizer"
+    c_compiler: "/opt/clang+llvm/bin/clang"
+    cxx_compiler: "/opt/clang+llvm/bin/clang++"
   tasks:
   - name: compile_test
   - name: long-running-tests
-    distros:
-    - ubuntu2004-small
 
-- name: ubuntu2004-tsan
-  display_name: "Ubuntu 20.04 x86_64 (Clang 11 TSAN)"
-  run_on: ubuntu2004-large
+- name: ubuntu-asan-x86_64
+  display_name: "Ubuntu (ASAN x86_64)"
+  run_on: ubuntu2204-large
   expansions:
-    clang_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/clang%2Bllvm-11.0.0-x86_64-linux-gnu-ubuntu-20.04.tar.xz"
+    fetch_missing_dependencies: On
     cmake_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/cmake-3.26.3-linux-x86_64.tar.gz"
     cmake_bindir: "./cmake_binaries/bin"
+    clang_url: "https://s3.amazonaws.com/static.realm.io/evergreen-assets/clang%2Bllvm-18.1.2-x86_64-linux-gnu.tar.xz"
+    clang_sha256_sum: "feab4b0f5fba325bfe0f4245710fd19fd74f813f44b5e81eda794f4f75bca343"
+    c_compiler: "./clang_binaries/bin/clang"
+    cxx_compiler: "./clang_binaries/bin/clang++"
+    fetch_missing_dependencies: On
+    enable_asan: On
+    cmake_build_type: RelWithDebInfo
+  tasks:
+  - name: compile_test_coverage
+  - name: finalize_coverage_data
```

Comment:
I don't think we need another finalize_coverage_data task, isn't it sufficient to just have one?

Reply:
Good catch! I think I copy/pasted something wrong here.